File: bmc/custompy/loan_application.py


import frappe


@frappe.whitelist()
def make_credit_committee_form(self):
	c_form = frappe.new_doc("Credit Committee Form")
	c_form.loan_application = self.name
	c_form.applicant_name = self.applicant_name
	c_form.types_of_loan = self.loan_kind_
	c_form.number_of_renewals = self.number_of_renewals
	c_form.url = self.url
	c_form.save(ignore_permissions=True)
	
	if c_form.name:
		loan_app = frappe.get_doc("Loan Application", self.name)
		loan_app.credit_committee_form = c_form.name
		loan_app.save(ignore_permission=True)

File: bmc/customjs/loan_application.js
frappe.ui.form.on('Loan Application', {
    refresh(frm){
        if (frm.doc.docstatus == 1) {
            frm.add_custom_button("Make Credit Committee Form", function () {
                make_credit_committee_form(frm);
            }).addClass("btn-primary");
        }
    }
});

let make_credit_committee_form = function (frm) {
	var doc = frm.doc;
	if (!doc.credit_committee_form) {
		return frappe.call({
			method: "bmc.custompy.loan_application.make_credit_committee_form",
            doc: cur_frm.doc,
			callback: function () {
				frappe.set_route(
					'List', 'Credit Committee Form', {
						"Credit Committee Form.loan_application": frm.doc.name
					}
				);
			},
			freeze: true,
			freeze_message: __("Creating Credit Committee Form............")
		});
	}
};